pkgs/tools/typesetting/tex/opendetex/default.nix

```diff
@@ -0,0 +1,46 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, flex
+, testers
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "opendetex";
+  version = "2.8.9";
+
+  src = fetchFromGitHub {
+    owner = "pkubowicz";
+    repo = "opendetex";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-8mwXsJImi8jESR87VA9NsUIvH52JJbbrjzpUdKPI+ls=";
+  };
+
+  postPatch = ''
+    substituteInPlace Makefile \
+      --replace "install -c -m 775 -g staff" "install -c -m 775" \
+      --replace "/usr/local" "$out"
+  '';
+
+  nativeBuildInputs = [
+    flex
+  ];
+
+  preInstall = ''
+    mkdir -p $out/share/man $out/bin
+  '';
+
+  passthru.tests.version = testers.testVersion {
+    package = finalAttrs.finalPackage;
+    command = "detex -v";
+  };
+
+  meta = with lib; {
+    homepage = "https://github.com/pkubowicz/opendetex";
+    description = "Improved version of Detex - tool for extracting plain text from TeX and LaTeX sources";
+    license = licenses.ncsa;
+    maintainers = with maintainers; [ fgaz ];
+    platforms = platforms.all;
+    mainProgram = "detex";
```

Comment:
Unfortunately most schemes of `texlive` contain a `detex` binary already, so you won't be able to install `opendetex` and `texlive` at the same time. Not sure if there is a solution to this problem – the author of the package has decided to use the same name, so we are stuck with the conflict.

Reply:
maybe add `priority = -1;` here. If someone installs texlive and opendetex into the same profile, they will almost certainly want detex to be from opendetex.